### ISSUE TITLE
多方状态同步的问题

### ISSUE THREAD

**Issue by killomsc** [secretflow/kuscia#472]
### Issue Type

Api Usage

### Search for existing issues similar to yours

Yes

### Kuscia Version

0.9

### Question Details

```shell
我们在尝试增加横向算法组件，涉及到arbiter的参与，但在kuscia调度的时候，发现协作方以及arbiter只有发起方与自身的状态，导致任务无法调度。如下：

发起方有三方的状态。
status:
  approveStatus:
    arbiter: JobAccepted
    b501d1f3b3424abe8d3d15357308ac94: JobAccepted
    e6a4ffdbeb584d6b8ea6c91332134e76: JobAccepted
  completionTime: "2024-12-23T08:09:25Z"
  conditions:
  - lastTransitionTime: "2024-12-23T08:04:25Z"
    status: "True"
    type: JobValidated
  lastReconcileTime: "2024-12-23T08:09:25Z"
  phase: Failed
  stageStatus:
    arbiter: JobCreateStageSucceeded
    b501d1f3b3424abe8d3d15357308ac94: JobCreateStageSucceeded
    e6a4ffdbeb584d6b8ea6c91332134e76: JobCreateStageSucceeded

其他两方只有自身以及发起方的状态。
status:
  approveStatus:
    b501d1f3b3424abe8d3d15357308ac94: JobAccepted
    e6a4ffdbeb584d6b8ea6c91332134e76: JobAccepted
  conditions:
  - lastTransitionTime: "2024-12-23T09:27:46Z"
    status: "True"
    type: JobValidated
  lastReconcileTime: "2024-12-23T09:27:46Z"
  phase: AwaitingApproval
  stageStatus:
    b501d1f3b3424abe8d3d15357308ac94: JobCreateStageSucceeded
    e6a4ffdbeb584d6b8ea6c91332134e76: JobCreateStageSucceeded
  startTime: "2024-12-23T09:27:46Z"
```

**Comment by BrainWH** [comment]
您好，可以提供以下信息：
1. 当前的 secretflow 的版本信息
2. 进入容器内部， 看一下 /home/kuscia/var/stdout/pods/alice_xxxx/secretflow/0.log

**Comment by killomsc** [comment]
算法容器还没调度起来，所以没有secretflow的日志。
我大概描述下，涉及三方
发起方的domain:b501d1f3b3424abe8d3d15357308ac94
合作方的domain:e6a4ffdbeb584d6b8ea6c91332134e76
arbiter的domain:arbiter

通过发起方的kuscia api，发起job任务。
job的parties有上述三个参与方
任务提交后，算法任务没有正常执行，看了下各方的job状态。

发起方：b501d1f3b3424abe8d3d15357308ac94
status:
  approveStatus:
    arbiter: JobAccepted
    b501d1f3b3424abe8d3d15357308ac94: JobAccepted
    e6a4ffdbeb584d6b8ea6c91332134e76: JobAccepted
  conditions:
  - lastTransitionTime: "2024-12-24T06:02:34Z"
    status: "True"
    type: JobValidated
  lastReconcileTime: "2024-12-24T06:02:35Z"
  phase: Running
  stageStatus:
    arbiter: JobCreateStageSucceeded
    b501d1f3b3424abe8d3d15357308ac94: JobCreateStageSucceeded
    e6a4ffdbeb584d6b8ea6c91332134e76: JobCreateStageSucceeded
  startTime: "2024-12-24T06:02:34Z"
  taskStatus:
    homo-merge-5dde1b437cbe5d3a5ae6594bba242fba: Pending


协作方：e6a4ffdbeb584d6b8ea6c91332134e76
status:
  approveStatus:
    b501d1f3b3424abe8d3d15357308ac94: JobAccepted
    e6a4ffdbeb584d6b8ea6c91332134e76: JobAccepted
  conditions:
  - lastTransitionTime: "2024-12-24T06:02:34Z"
    status: "True"
    type: JobValidated
  lastReconcileTime: "2024-12-24T06:02:35Z"
  phase: AwaitingApproval
  stageStatus:
    b501d1f3b3424abe8d3d15357308ac94: JobCreateStageSucceeded
    e6a4ffdbeb584d6b8ea6c91332134e76: JobCreateStageSucceeded
  startTime: "2024-12-24T06:02:34Z"


arbiter方：
status:
  approveStatus:
    arbiter: JobAccepted
    b501d1f3b3424abe8d3d15357308ac94: JobAccepted
  conditions:
  - lastTransitionTime: "2024-12-24T06:02:34Z"
    status: "True"
    type: JobValidated
  lastReconcileTime: "2024-12-24T06:03:52Z"
  phase: AwaitingApproval
  stageStatus:
    arbiter: JobCreateStageSucceeded
    b501d1f3b3424abe8d3d15357308ac94: JobCreateStageSucceeded
  startTime: "2024-12-24T06:02:34Z"

发起方，能看到三方的approveStatus，此时  phase: Running
协作方，只能看到自己以及发起方的approveStatus，此时phase: AwaitingApproval
arbiter，只能看到自己以及发起方的approveStatus，此时phase: AwaitingApproval

采用p2p的部署模式，默认情况下，job是会自动审批的，但目前协作方和arbiter的job一直卡在审批这里，没有进入Pending。
大概看了下approval代码，进入Pending，需要所有参与方都Accepted，但协作方e6a4ffdbeb584d6b8ea6c91332134e76没有arbiter的approveStatus，arbiter方没有e6a4ffdbeb584d6b8ea6c91332134e76的approveStatus，所以导致协作方的job的状态一直没有进入Pending。而发起方的job是正常的，因为能看到所有参与方的status。
	// Only inter connection job need approval
	// the logic of handle awaiting approval is no different between  self as initiator or as partner
	// all partner have approval accept
	if ok, _ := h.allApprovalAccept(job); ok {
		// set AwaitingApproval --> Pending
		job.Status.Phase = kusciaapisv1alpha1.KusciaJobPending
		needUpdateStatus = true
	}


那么问题来了，就是三方以上的时候，为什么协作方，只能看到自己以及发起方的approveStatus，而看不到arbiter的approveStatus，这个informer的同步应该怎么设置？或者说kuscia的代码上是否支持三方以上的调度？

麻烦协助看看，谢谢

**Comment by BrainWH** [comment]
可以提供一下 job 的配置信息

**Comment by killomsc** [comment]
下面是发起方的job信息，@BrainWH 

apiVersion: kuscia.secretflow/v1alpha1
kind: KusciaJob
metadata:
  annotations:
    kuscia.secretflow/initiator: b501d1f3b3424abe8d3d15357308ac94
    kuscia.secretflow/interconn-kuscia-parties: arbiter_e6a4ffdbeb584d6b8ea6c91332134e76
    kuscia.secretflow/interconn-self-parties: b501d1f3b3424abe8d3d15357308ac94
    kuscia.secretflow/self-cluster-as-initiator: "true"
  creationTimestamp: "2024-12-23T09:27:46Z"
  generation: 1
  name: ac8c56dd65b84202a687a68575b3c755
  namespace: cross-domain
  resourceVersion: "67284195"
  uid: dbdca43f-0e80-423f-970d-4071151d8a79
spec:
  initiator: b501d1f3b3424abe8d3d15357308ac94
  maxParallelism: 3
  scheduleMode: BestEffort
  tasks:
  - alias: homo-merge-508a46fb3d41974c420fb6a8223516fc
    appImage: secretflow-image
    parties:
    - domainID: b501d1f3b3424abe8d3d15357308ac94
    - domainID: e6a4ffdbeb584d6b8ea6c91332134e76
    - domainID: arbiter
    taskID: homo-merge-508a46fb3d41974c420fb6a8223516fc
    taskInputConfig: |-
      {
        "sf_datasource_config": {
          "b501d1f3b3424abe8d3d15357308ac94": {
            "id": "minio-data-source"
          },
          "arbiter": {
            "id": "minio-data-source"
          },
          "e6a4ffdbeb584d6b8ea6c91332134e76": {
            "id": "minio-data-source"
          }
        },
        "sf_cluster_desc": {
          "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76", "arbiter"],
          "devices": [{
            "name": "spu",
            "type": "spu",
            "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76"],
            "config": "{\"runtime_config\":{\"protocol\":\"SEMI2K\",\"field\":\"FM128\"},\"link_desc\":{\"connect_retry_times\":60,\"connect_retry_interval_ms\":1000,\"brpc_channel_protocol\":\"http\",\"brpc_channel_connection_type\":\"pooled\",\"recv_timeout_ms\":1200000,\"http_timeout_ms\":1200000}}"
          }, {
            "name": "heu",
            "type": "heu",
            "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76"],
            "config": "{\"mode\": \"PHEU\", \"schema\": \"paillier\", \"key_size\": 2048}"
          }, {
            "name": "agg",
            "type": "agg",
            "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76", "arbiter"],
            "config": "{\"server\":\"b501d1f3b3424abe8d3d15357308ac94\",\"clients\":[\"b501d1f3b3424abe8d3d15357308ac94\",\"e6a4ffdbeb584d6b8ea6c91332134e76\",\"arbiter\"],\"type\":\"secure\"}"
          }],
          "ray_fed_config": {
            "cross_silo_comm_backend": "brpc_link"
          }
        },
        "sf_node_eval_param": {
          "domain": "data_prep",
          "name": "homo_merge",
          "version": "0.0.1",
          "inputs": [{
            "type": "sf.table.individual",
            "meta": {
              "@type": "type.googleapis.com/secretflow.spec.v1.IndividualTable"
            },
            "data_refs": [{
              "uri": "table_data_set/ecb1328ef5434982af15fa7c77f042f5.csv",
              "party": "e6a4ffdbeb584d6b8ea6c91332134e76",
              "format": "csv"
            }]
          }, {
            "type": "sf.table.individual",
            "meta": {
              "@type": "type.googleapis.com/secretflow.spec.v1.IndividualTable"
            },
            "data_refs": [{
              "uri": "table_data_set/57328ff361874946b9fea09bcb8ff78c.csv",
              "party": "b501d1f3b3424abe8d3d15357308ac94",
              "format": "csv"
            }]
          }]
        },
        "sf_output_uris": ["ac8c56dd65b84202a687a68575b3c755-t0-homo-merge-out0"],
        "sf_input_ids": ["ecb1328ef5434982af15fa7c77f042f5", "57328ff361874946b9fea09bcb8ff78c"],
        "sf_output_ids": ["ac8c56dd65b84202a687a68575b3c755-t0-homo-merge-out0"]
      }
    tolerable: false
  - alias: homo-train-test-split-4cf43ae791eb054f75832a4803b9724a
    appImage: secretflow-image
    dependencies:
    - homo-merge-508a46fb3d41974c420fb6a8223516fc
    parties:
    - domainID: b501d1f3b3424abe8d3d15357308ac94
    - domainID: e6a4ffdbeb584d6b8ea6c91332134e76
    - domainID: arbiter
    taskID: homo-train-test-split-4cf43ae791eb054f75832a4803b9724a
    taskInputConfig: |-
      {
        "sf_datasource_config": {
          "b501d1f3b3424abe8d3d15357308ac94": {
            "id": "minio-data-source"
          },
          "arbiter": {
            "id": "minio-data-source"
          },
          "e6a4ffdbeb584d6b8ea6c91332134e76": {
            "id": "minio-data-source"
          }
        },
        "sf_cluster_desc": {
          "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76", "arbiter"],
          "devices": [{
            "name": "spu",
            "type": "spu",
            "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76"],
            "config": "{\"runtime_config\":{\"protocol\":\"SEMI2K\",\"field\":\"FM128\"},\"link_desc\":{\"connect_retry_times\":60,\"connect_retry_interval_ms\":1000,\"brpc_channel_protocol\":\"http\",\"brpc_channel_connection_type\":\"pooled\",\"recv_timeout_ms\":1200000,\"http_timeout_ms\":1200000}}"
          }, {
            "name": "heu",
            "type": "heu",
            "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76"],
            "config": "{\"mode\": \"PHEU\", \"schema\": \"paillier\", \"key_size\": 2048}"
          }, {
            "name": "agg",
            "type": "agg",
            "parties": ["b501d1f3b3424abe8d3d15357308ac94", "e6a4ffdbeb584d6b8ea6c91332134e76", "arbiter"],
            "config": "{\"server\":\"b501d1f3b3424abe8d3d15357308ac94\",\"clients\":[\"b501d1f3b3424abe8d3d15357308ac94\",\"e6a4ffdbeb584d6b8ea6c91332134e76\",\"arbiter\"],\"type\":\"secure\"}"
          }],
          "ray_fed_config": {
            "cross_silo_comm_backend": "brpc_link"
          }
        },
        "sf_node_eval_param": {
          "domain": "data_prep",
          "name": "homo_train_test_split",
          "version": "0.0.1",
          "attr_paths": ["train_size", "test_size", "random_state", "shuffle"],
          "attrs": [{
            "f": 0.7
          }, {
            "f": 0.3
          }, {
            "i64": "1024"
          }, {
            "b": true
          }]
        },
        "sf_output_uris": ["ac8c56dd65b84202a687a68575b3c755-t1-homo-train-test-split-out0", "ac8c56dd65b84202a687a68575b3c755-t1-homo-train-test-split-out1", "ac8c56dd65b84202a687a68575b3c755-t1-homo-train-test-split-out2"],
        "sf_input_ids": ["ac8c56dd65b84202a687a68575b3c755-t0-homo-merge-out0"],
        "sf_output_ids": ["ac8c56dd65b84202a687a68575b3c755-t1-homo-train-test-split-out0", "ac8c56dd65b84202a687a68575b3c755-t1-homo-train-test-split-out1", "ac8c56dd65b84202a687a68575b3c755-t1-homo-train-test-split-out2"]
      }
    tolerable: false
status:
  approveStatus:
    arbiter: JobAccepted
    b501d1f3b3424abe8d3d15357308ac94: JobAccepted
    e6a4ffdbeb584d6b8ea6c91332134e76: JobAccepted
  completionTime: "2024-12-23T09:32:46Z"
  conditions:
  - lastTransitionTime: "2024-12-23T09:27:46Z"
    status: "True"
    type: JobValidated
  lastReconcileTime: "2024-12-23T09:32:46Z"
  phase: Failed
  stageStatus:
    arbiter: JobCreateStageSucceeded
    b501d1f3b3424abe8d3d15357308ac94: JobCreateStageSucceeded
    e6a4ffdbeb584d6b8ea6c91332134e76: JobCreateStageSucceeded
  startTime: "2024-12-23T09:27:46Z"
  taskStatus:
    homo-merge-508a46fb3d41974c420fb6a8223516fc: Failed

**Comment by BrainWH** [comment]
您好，可以提供一下 kuscia 的日志

**Comment by killomsc** [comment]
[kuscia.log](https://github.com/user-attachments/files/18238102/kuscia.log)
发起方的kuscia.log

**Comment by BrainWH** [comment]
先检查一下可用资源是多少？

**Comment by killomsc** [comment]
是有这么一个错误日志，然后我们最初也是怀疑是资源的情况，但看了资源是足够的。是不是协作方没调度起来，就认为是资源不够？
Can't schedule the domain b501d1f3b3424abe8d3d15357308ac94 pod homo-merge-30ee6d478e79a5a48c59b6bf659f620a-0 because task resource homo-merge-30ee6d478e79a5a48c59b6bf659f620a-6455825e473a status phase isn't schedulable, domain [e6a4ffdbeb584d6b8ea6c91332134e76,arbiter] can not reserve resources for pods

**Comment by killomsc** [comment]
看协作方还卡在待审批的状态。

**Comment by BrainWH** [comment]
可以使用 kuscia 0.13.0b0, 这个版本对于三方的支持更加完善

**Comment by killomsc** [comment]
好的，我看看

**Comment by killomsc** [comment]
升级到新版本后，此问题解决了
